lib/wmi-lite/wmi.rb

```diff
@@ -17,14 +17,23 @@
 #
 
 require 'win32ole' if RUBY_PLATFORM =~ /mswin|mingw32|windows/
+require 'wmi-lite/exceptions'
 require 'wmi-lite/wmi_instance'
-require 'wmi-lite/wmi_exception'
+require 'timeout'
 
 module WmiLite
   class Wmi
     def initialize(namespace = nil)
       @namespace = namespace.nil? ? 'root/cimv2' : namespace
       @connection = nil
+
+      # This timeout should only be reached if there is something wrong 
+      # with the WMI subsystem causing it to hang requests
+      @timeout = 120
```

Comment:
Let's keep the default as-is so we don't affect other users of the gem, and allow callers like Chef and Ohai to opt-in to the timeout.